modules/programs/astroid.nix

```diff
@@ -1,129 +1,131 @@
 
-
 in
-{
 
+{
   options = {
     programs.astroid = {
       enable = mkEnableOption "Astroid";
 
       pollScript = mkOption {
         type = types.str;
         default = "";
-        example = ''
-          mbsync gmail
-        '';
+        example = "mbsync gmail";
         description = ''
           Script to run to fetch/update mails.
         '';
       };
 
       externalEditor = mkOption {
-        type = types.str;
-        default = ''
-          gvim -c 'set ft=mail' '+set fileencoding=utf-8' '+set ff=unix' '+set enc=utf-8' '+set fo+=w' %1
-        '';
-        # converts it into json that can be appended to the config
+        type = types.nullOr types.str;
+        default = null;
+        # Converts it into JSON that can be merged into the configuration.
         apply = cmd:
-          {
+          optionalAttrs (cmd != null) {
             editor = {
               "external_editor" = "true";
               "cmd" = cmd;
             };
           };
-
-        example = ''
-          nvim-qt -- -c 'set ft=mail' '+set fileencoding=utf-8' '+set ff=unix' '+set enc=utf-8' '+set fo+=w' %1
-        '';
+        example = "nvim-qt -- -c 'set ft=mail' '+set fileencoding=utf-8' '+set ff=unix' '+set enc=utf-8' '+set fo+=w' %1";
         description = ''
-          You can use %1 / %2 / %3 to refer respectively to:
-          - filename
-          - servername
-          - socketid
-          See <link xlink:href='https://github.com/astroidmail/astroid/wiki/Customizing-editor' />
+          You can use <code>%1</code>, <code>%2</code>, and
+          <code>%3</code> to refer respectively to:
+          <orderedlist numeration="arabic">
+            <listitem><para>file name</para></listitem>
+            <listitem><para>server name</para></listitem>
+            <listitem><para>socket ID</para></listitem>
+          </orderedlist>
+          See <link xlink:href='https://github.com/astroidmail/astroid/wiki/Customizing-editor' />.
         '';
       };
 
       extraConfig = mkOption {
         type = types.attrs;
         default = {};
-        example = literalExample ''
-          {
-            poll.interval = 0;
-          }
-        '';
+        example = { poll.interval = 0; };
         description = ''
-          JSON config that will override the default astroid configuration.
+          JSON config that will override the default Astroid configuration.
         '';
       };
     };
   };
 
   config = mkIf cfg.enable {
-
     home.packages =  [ pkgs.astroid ];
 
-    xdg.configFile."astroid/config".source = pkgs.runCommandNoCC "out.json"
-    {x = configFile astroidAccounts; } ''echo "$x" | ${pkgs.jq}/bin/jq . > $out'';
+    xdg.configFile."astroid/config".source =
+      pkgs.runCommand "out.json"
+        {
+          json = configFile astroidAccounts;
+          preferLocalBuild = true;
+          allowSubstitutes = false;
+        }
+        ''
+          echo -n "$json" | ${pkgs.jq}/bin/jq . > $out
+        '';
```

Comment:
I wonder if some of these couldn't be factored in lib/utils.nix or similar.

Reply:
Yeah, if more modules need to write pretty JSON then I think this is a perfect candidate for abstraction.